File: packages/core/src/__tests__/fixtures/crosslang-sample.ts
/**
 * Cross-language compatibility test sample (TypeScript)
 *
 * This file contains simple functions to test that all analyzers
 * produce compatible output structures across TypeScript, Python, and Go.
 */

// Simple function with control flow
export function simpleControl(x: number): string {
  if (x > 0) {
    return 'positive';
  } else if (x < 0) {
    return 'negative';
  }
  return 'zero';
}

// Function with nesting
export function nestedLoop(arr: number[][]): number {
  let sum = 0;
  for (const row of arr) {
    for (const val of row) {
      if (val > 0) {
        sum += val;
      }
    }
  }
  return sum;
}

// Function with state mutations
export function stateMutation(): number {
  let state = 0;
  let status = 'idle';

  status = 'processing';
  state = 1;

  status = 'done';
  state = 2;

  return state;
}

// Async function
export async function asyncFunction(): Promise<string> {
  const response = await fetch('/api/data');
  const data = await response.json();
  return data.message;
}

// Function with coupling (side effects)
export function coupledFunction(msg: string): void {
  console.log(msg);
  localStorage.setItem('lastMsg', msg);
}

File: packages/core/src/__tests__/fixtures/crosslang_sample.py
"""
Cross-language compatibility test sample (Python)

This file contains simple functions to test that all analyzers
produce compatible output structures across TypeScript, Python, and Go.
"""


def simple_control(x: int) -> str:
    """Simple function with control flow."""
    if x > 0:
        return "positive"
    elif x < 0:
        return "negative"
    return "zero"


def nested_loop(arr: list[list[int]]) -> int:
    """Function with nesting."""
    total = 0
    for row in arr:
        for val in row:
            if val > 0:
                total += val
    return total


def state_mutation() -> int:
    """Function with state mutations."""
    state = 0
    status = "idle"

    status = "processing"
    state = 1

    status = "done"
    state = 2

    return state


async def async_function() -> str:
    """Async function."""
    import aiohttp
    async with aiohttp.ClientSession() as session:
        async with session.get("/api/data") as response:
            data = await response.json()
            return data["message"]


def coupled_function(msg: str) -> None:
    """Function with coupling (side effects)."""
    print(msg)
    with open("last_msg.txt", "w") as f:
        f.write(msg)
